mappings/net/minecraft/block/entity/ViewerCountManager.mapping

```diff
@@ -1,36 +1,38 @@
-CLASS net/minecraft/class_5561 net/minecraft/block/entity/ChestStateManager
-	COMMENT Handles the viewer count for chest-like block entities.
+CLASS net/minecraft/class_5561 net/minecraft/block/entity/ViewerCountManager
+	COMMENT Handles the viewer count for container block entities, like chests,
+	COMMENT ender chests, and barrels.
 	FIELD field_27215 viewerCount I
+	FIELD field_31339 SCHEDULE_TICK_DELAY I
 	METHOD method_31678 getViewerCount ()I
 	METHOD method_31679 isPlayerViewing (Lnet/minecraft/class_1657;)Z
-		COMMENT Determines whether the given player is currently viewing this chest.
+		COMMENT Determines whether the given player is currently viewing this container.
 		ARG 1 player
 	METHOD method_31680 getInRangeViewerCount (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)I
 		ARG 1 world
 		ARG 2 pos
-	METHOD method_31681 onChestOpened (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)V
-		COMMENT Run when this chest is opened (when the viewer count becomes nonzero).
+	METHOD method_31681 onContainerOpen (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)V
+		COMMENT Run when this container is opened (when the viewer count becomes nonzero).
 		ARG 1 world
 		ARG 2 pos
 		ARG 3 state
 	METHOD method_31682 onInteracted (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;II)V
```

Comment:
Still, this actually updates the viewer count, even if open/close methods are not called.
```suggestion
	METHOD method_31682 onViewerCountUpdate (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;II)V
```